File: app/src/main/java/com/example/paper_slide/ui/imortfile/ImageAdapter.java
package com.example.paper_slide.ui.imortfile;

import android.annotation.SuppressLint;
import android.content.Context;
import android.content.Intent;
import android.net.Uri;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;
import com.bumptech.glide.Glide;
import com.example.paper_slide.R;

import java.util.ArrayList;

public class ImageAdapter extends RecyclerView.Adapter<ImageAdapter.viewHolder> {
    private ArrayList<ImageModel> list;
    private Context context;
    private viewHolder holder;
    private int position;

    public ImageAdapter(ArrayList<ImageModel> list, Context context) {
        this.list = list;
        this.context = context;
    }
    @NonNull
    @Override
    public viewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(context).inflate(
                R.layout.list_items, parent, false);
        return new viewHolder(view);
    }
    @Override
    public void onBindViewHolder(@NonNull viewHolder holder, @SuppressLint("RecyclerView") int position) {



        if (list.get(position).getPath() != null) {
       //     Glide.with(context).load(list.get(position).getPath()).into(holder.imageView);
        }


          holder.imageView.setOnClickListener(new View.OnClickListener() {
                @Override
                public void onClick(View view) {
                    String parseData = list.get(holder.getAdapterPosition()).getPath().toString();
                    context.startActivity(new Intent(context, FullScreenActivity.class)
                            .putExtra("parseData", parseData));
                }
            });



    }
    @Override
    public int getItemCount() {
        return list.size();
    }

    public class viewHolder extends RecyclerView.ViewHolder
    {
        ImageView imageView;
        public viewHolder(@NonNull View itemView) {
            super(itemView);
            imageView = itemView.findViewById(R.id.imageViewgallary);
        }
    }
}

File: app/src/main/java/com/example/paper_slide/ui/ppt_themes/ImageAdapter.kt
package com.example.paper_slide.ui.ppt_themes

import android.net.Uri
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import androidx.recyclerview.widget.RecyclerView
import com.example.paper_slide.R


class ImageAdapter(private var imageUris: List<Uri>) : RecyclerView.Adapter<ImageAdapter.ImageViewHolder>() {


    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ImageViewHolder {
        val view = LayoutInflater.from(parent.context).inflate(R.layout.theme_item, parent, false)
        return ImageViewHolder(view)
    }

    override fun onBindViewHolder(holder: ImageViewHolder, position: Int) {
        val imageUri = imageUris[position]
        holder.bind(imageUri)
    }

    override fun getItemCount(): Int {
        return imageUris.size
    }

    inner class ImageViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        private val imageView: ImageView = itemView.findViewById(R.id.imageView)

        fun bind(imageUri: Uri) {
            imageView.setImageURI(imageUri)
        }
    }

    /*fun setData(newData: List<Uri>) {
        imageUris.clear()
        imageUris.addAll(newData)
        notifyDataSetChanged()
    }*/
}